.changelog/25598.txt

```diff
@@ -0,0 +1,3 @@
+```release-note:improvement
+task environment: add NOMAD_UNIX_ADDR env var when using workload identity
```

Comment:
"when using workload identity" doesn't quite feel like the right way to describe this. It's _for use with_ workload identity. Maybe point out that it points to the [Task API socket](https://developer.hashicorp.com/nomad/api-docs/task-api)?